python/oneflow/nn/modules/global_cast.py

```diff
@@ -75,16 +77,56 @@ def global_to_global_op(
 
 
 def to_global_op(input, placement=None, sbp=None, **kwargs):
-    assert isinstance(input, Tensor)
-
-    if input.is_global:
-        return global_to_global_op(input=input, placement=placement, sbp=sbp, **kwargs)
-    else:
-        if "grad_sbp" in kwargs:
-            del kwargs["grad_sbp"]
-        return local_to_global_op(input=input, placement=placement, sbp=sbp, **kwargs)
+    r"""Converts the input to global if it is local, otherwise performs designated placement and/or sbp conversion.
+    
+    It performs the same conversion as :func:`oneflow.Tensor.to_global` to the tensor or every tensor in the state dict.
+
+    Note:
+        Both placement and sbp are required if the input is local, otherwise at least one of placement and sbp is required.
+
+    Args:
+        input (flow.Tensor/None/list/tuple/dict): the input that needs to be converted.
+        placement (flow.placement, optional):  the desired placement of the input. Default: None
+        sbp (flow.sbp.sbp or list/tuple of flow.sbp.sbp, optional): the desired sbp of the input. Default: None
+    
+    Returns:
+        The converted tensor / state dict.
+    """
+    input_tree = IONode(value=input)
+
+    def leaf_fn(node):
+        if isinstance(node._value, Tensor):
+            if node._value.is_global:
+                return global_to_global_op(input=node._value, placement=placement, sbp=sbp, **kwargs)
+            else:
+                if "grad_sbp" in kwargs:
+                    del kwargs["grad_sbp"]
+                return local_to_global_op(input=node._value, placement=placement, sbp=sbp, **kwargs)
+        else:
+            warnings.warn("Non-Tensor type: {} encountered, it will not be converted.".format(type(node._value)))
+            return node._value
+
+    mapped_input = input_tree.map_leaf(leaf_fn)
+    return mapped_input
 
 
 def to_local_op(input):
-    assert input.is_global, "input must be a global tensor!"
-    return flow._C.to_local(input)
+    r"""Returns the local component of the input.
+
+    It performs the same conversion as :func:`oneflow.Tensor.to_local` to the tensor(s) in the input.
+    
+    Returns:
+        The converted input.
+    """
+    input_tree = IONode(value=input)
+
+    def leaf_fn(node):
+        if isinstance(node._value, Tensor):
```

Comment:
Refer to  `to_global_tensor`, refactor this with `to_local_tensor`.

Reply:
Corrected